gcc/rust/ast/rust-ast-collector.h

```diff
@@ -33,13 +34,16 @@ class CollectItem
   enum class Kind
   {
     Comment,
+    InternalComment,
     Newline,
     Indentation,
     Token,
   };
 
   CollectItem (TokenPtr token) : token (token), kind (Kind::Token) {}
-  CollectItem (std::string comment) : comment (comment), kind (Kind::Comment) {}
+  CollectItem (std::string comment, bool internal = false)
```

Comment:
I would avoid using the boolean here and expose a new enum instead, which would make the code clearer

```c++
enum class Comment {
    Regular,
    Internal,
};

CollectItem (std::string comment, Comment kind = Comment::Regular) : comment(comment), kind(kind == Comment::Internal ? Kind::InternalComment : Kind::Comment) { ... }
```